pkg/policy/tree.go

```diff
@@ -89,45 +89,58 @@ func canConsume(root *Node, ctx *SearchContext) api.ConsumableDecision {
 
 // AllowsRLocked checks the ConsumableDecision of the given `SearchContext`.
 // Must be called with the Mutex's tree at least RLocked.
```

Comment:
Why not have a wrapper function called AllowsRLocked and change this function to be allowsRLocked and then in AllowsRLocked, lock the tree? We should be doing the locking if it is required by the function.